File: exercism/java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Etl {
    Map<String, Integer> transform(Map<Integer, List<String>> old) {
        Map<String, Integer> newMap = new HashMap<>();
        old.forEach(((valueType, keyTypes) ->
                keyTypes.forEach(keyType ->
                        newMap.put(keyType.toLowerCase(), valueType)
                )
        ));
        return newMap;
    }
}

File: exercism/scala/etl/src/main/scala/Etl.scala


object Etl {
  def transform(scoreMap: Map[Int, Seq[String]]): Map[String, Int] = {
    val reverseMap = for ((k,v) <- scoreMap) yield for (c <- v) yield (c.toLowerCase(), k)
    reverseMap.flatten.toMap
  }
}
